### DISCUSSION TITLE
v4.1.0

### DISCUSSION THREAD

**Discussion by mostypc123** [mostypc123/XediX#47] [Announcements]
## Major Features 🌟
- 🐧 Linux support with installer script
- 📊 Status bar with debug messages
- 🧩 Extension system with documentation
- 🎨 Theme support
- 📝 Requirements.txt template generation
- 🔔 GitHub Issue/PR notifications
- ✨ UI improvements and optimizations

## What's Changed 📋
* **v4.1.alpha.1**
  * 📁 Images by @mostypc123 in https://github.com/mostypc123/XediX/pull/18
    * Created `/images` folder for better file structure.
  * 🚀 Version 4.1.alpha.1 by @mostypc123 in https://github.com/mostypc123/XediX/pull/19
    * 📊 Added a status bar with basic debug messages. **In Beta**
    * ⌨️ Auto completion of `()`, `{}`, `[]` and `""` with `''`.
    * 🔧 When you created a new file, it was showing the whole filepath. Now it's showing just the filename. 
* **v4.1.alpha.2**
  * 🗂️ Filetree fixing by @mostypc123 in https://github.com/mostypc123/XediX/pull/20
    * Improved file structure just again
  * ⚡ Update by @mostypc123 in https://github.com/mostypc123/XediX/pull/21
    * Optimization
    * 📝 `README.md` and `SECURITY.md` improvements
  * 🎨 Syntax highlighting improvements by @mostypc123 in https://github.com/mostypc123/XediX/pull/22
    * Improved the syntax highlighting
  * 🧩 Extensions by @mostypc123 in https://github.com/mostypc123/XediX/pull/23
    * To extend XediX's functionality. To add a official extension to our list, visit `mostypc123/xedix-extensions`.
    * Documentation is available at Github Wikies.
* **[v4.1.alpha.3](https://github.com/mostypc123/XediX/compare/v4.1.alpha.2...v4.1.alpha.3)**
  * 🔄 Minor changes and improvements (see comparison)
* **[v4.1.alpha.4](https://github.com/mostypc123/XediX/compare/v4.1.alpha.3...v4.1.alpha.4)**
  * 🔄 Minor changes and improvements (see comparison)
* **v4.1.alpha.5**
  * 💻 Commands, to run XediX with the terminal by @mostypc123 in https://github.com/mostypc123/XediX/pull/27
    * Run `open.bat` and get prompted for a filename, and it autommactly opens XediX on that file.
  * ✨ Some slight improvements by @mostypc123 in https://github.com/mostypc123/XediX/pull/28
    * ↕️ Jump to line
    * 📊 Status bar
    * ❓ Help menu
  * 🌐 README Translations by @mostypc123 in https://github.com/mostypc123/XediX/pull/29
    * Translated the README to slovak
* **v4.1.alpha.6-7**
  * 🎨 Themes by @mostypc123 in https://github.com/mostypc123/XediX/pull/32
    * Added some themes
  * 🪟 Pywinstyles settings by @mostypc123 in https://github.com/mostypc123/XediX/pull/34
    _set the pywinstyles value, removed in #37_
* **v4.1.alpha.8**
  * 🎯 Enhancement: Basic UI improvements for the editor by @divyanshudhruv in https://github.com/mostypc123/XediX/pull/37
    * Improved the UI. **HUGE** thanks to @divyanshudhruv for contributing.
  * 📦 Requirements.txt Generation by @mostypc123 in https://github.com/mostypc123/XediX/pull/38
    * Tired of writing `requirements.txt` on your own? Get a template based on the code.
* **Last fixes/features before this release**
  * 🐛 issue 42 fixes by @mostypc123 in https://github.com/mostypc123/XediX/pull/43
    * Some additional features, check the feature request at #42 by @Dev-Hassaan
  * 🔔 [✨ feat: Github Issue/PR notifications](https://github.com/mostypc123/XediX/commit/f9bc9f0fd4007b7158f799cea71251fec8f437c8) commit by @mostypc123

**Full Changelog**: https://github.com/mostypc123/XediX/compare/v4.0.5...v4.1.0

## Installation 💿
*🐧 Linux and 🪟 Windows supported*

### Windows 🪟
> [!IMPORTANT]
> Yes, your antivirus/Widnows itself would say that it is a virus.
> The code is open source, you are more that welcome to read the source line by line.
> The antivirus detection happens because the .exe is not authored by a trusted company.

📥 Download the `XediX4.1Windows.zip` file, extract it and finally run the file `main.exe`.

### Linux 🐧
> [!TIP]
> To avoid the warning below, move all files from the `linux` folder to the main directory before running the installer.

📥 Download the `XediX4.1Linux.tar.xz` file, unzip it and open the terminal in the unzipped folder.
Run these commands:
```sh
# change the directory to linux installation folder
cd linux
# allow linuxinst.sh to execute
chmod +x linuxinst.sh
# run linuxinst.sh
linuxinst.sh
```
> [!WARNING]
> If you encounter a `FileNotFoundError`, move linuxrun.sh to the main directory and run: `python3 main.py`

### Windows 🪟
- Windows 7 or later
 - ✅ Is tested on Windows 11 and 10.
- 🐍 Python 3.x
 - ✅ Is tested on Python 3.8

### Linux 🐧
- Any modern Linux distribution
 - ✅ Is tested on Ubuntu 24.12
- 🐍 Python 3.x
 - ✅ Is tested on Python 3.12
- 💻 Terminal access and root permissions
 - 🔒 Uses `sudo apt install`. If you think the script is a virus, you are more than welcome to read the script.` it doesnt rm -rf ur pc bruh`

## Before you leave... 👋 - Developer's note
![image](https://github.com/user-attachments/assets/b4db0d93-e95b-40ea-adbe-701adc1155d6)
_xedix 4.1 on windows_

It's the start of 2025 🎉, and we did recode XediX from the scratch 🔄, added Linux support for the first time 🐧 and published two very large releases 🚀. 7 new stars ⭐. Sounds crazy to me 🤯.

I also made a 2024 Recap, will be updated because I was working on it even after making the recap. Anyways, check it out [here](https://xedix.w3spaces.com/recap24.html).

![image](https://github.com/user-attachments/assets/1bfbf1d6-c066-4f1e-b1ff-de8d98778329)
_first run of xedix on linux_

![image](https://github.com/user-attachments/assets/9f69c40e-8ce6-4766-a964-03f279038d45)
_second run of xedix on linux_

To celebrate this milestone 🏆, I'm releasing the oldest version of XediX I found on my PC 💾.
It's ugly 😅, barely works 🫠, but hey – XediX2 is now downloadable below. Have fun installing dependencies 🤣.

I also wanted to share the first attempt to make it, but i only found a screenshot. It worked like a notes app, and i transformed into XediX later.
![image (2)](https://github.com/user-attachments/assets/cdbac709-0f75-4c6e-aa35-d462e09c1a33)
It's interesting how that screenshot is almost 3 years old! 

<hr /><em>This discussion was created from the release <a href='https://github.com/mostypc123/XediX/releases/tag/v4.1.0'>v4.1.0</a>.</em>

**Comment by divyanshudhruv** [comment]
🔥🔥
